### DISCUSSION TITLE
Nintendo Switch Support

### DISCUSSION THREAD

**Discussion by Fancy2209** [airsdk/Adobe-Runtime-Support#3099] [General]
On https://github.com/airsdk/Adobe-Runtime-Support/discussions/3070#discussioncomment-8527718, it was mentioned that Nintendo Switch support for the AIR SDK is beeing looked into, could we know more about this?

**Comment by ajwfrost** [comment]
The current thoughts are that we may be able to create a version of Adobe AIR that supports some custom build/packaging process that would allow people to package their games for the Nintendo Switch. In the proposed scenario, developers would still need to sign up to the Nintendo developer mechanism and set themselves up, get their game approved, etc - but they would be given the option of using the custom AIR SDK during the development process. 

Currently there are some potential complexities though, because some of the limitations we have in our agreement from Adobe may make it incompatible with the way this is set up to work.. we're trying to find a route forwards for this, will have to see how it goes.

thanks

**Reply by Fancy2209** [reply]
Ah thanks, was just really curious about what you meant by that because if it is possible somehow for it to happen, it'd be huge (as consoles have never had flash support to port games to them), so I thought it was weird how it was a footnote on a reply, but this makes sense. I really hope these incompatibilities with the agreements can be sorted out!

**Reply by Fancy2209** [reply]
It's been a year, any updates on this or still trying to sort the legal issues out?

**Reply by ajwfrost** [reply]
We couldn't get a generic agreement with Nintendo that would allow us to become a 'framework developer' in their program, such that AIR SDK could be offered from within their developer portal. But they were happy with us doing this separately on a case-by-case basis i.e. if you are accepted onto the Nintendo developer program, we would then need to enter a separate agreement with you so that you could use the custom AIR SDK to build your app (as we can't deliver that publicly) and then you would need to go through all the normal test/acceptance processes with that.

So we did build a Nintendo Switch "platform" of the AIR runtime last year, but we haven't yet got hold of a development device in order to test this on, so it's a bit alpha-quality at the moment. We were waiting to see the "Switch 2" updates - although it sounded like there's backwards compatibility there. But it's probably nearing the stage where we can push to get a lead application signed up and do the final updates needed to get something working on the test device..

(If you already have a development Switch platform and an agreement with Nintendo, we'd love to hear from you!)

**Reply by Fancy2209** [reply]
I am not a licensed developer yet but will definetely contact once I am ready to release an app on the platform!
Will spread the word though as I think I know people who have a agreement/dev hardware and want to publish their apps on Switch.
Thanks for the update, really good to see AIR on Consoles is finally happening!

By the way yes, Switch 2 is retro compatible as confirmed at the end of the Annoucement Trailer for the console

**Reply by neutronized** [reply]
Super interested about a potential Nintendo Switch support for the AIR SDK.

If that can be of any help, I am a licensed Nintendo developer and own a Nintendo Switch devkit.

**Reply by Mark-LaCroix** [reply]
@ajwfrost 

I am also a licensed Nintendo Switch developer with access to dev hardware and have previously published a few games on the eShop (mostly using Unity). I have also ported AIR games to Haxe/OpenFL and deployed it to Switch, but I would welcome the opportunity to use the actual AIR runtime.

I'd love to help out with on-device testing and can serve as a test-case for you with a small AIR 50.0 hobby project that I've been working on.

You can contact me at mark@noblerobot.com

**Reply by DeaavhTV** [reply]
I've been writing a game engine for quite some time now, I would really enjoy this implementation alongside GPU compute with support for atomics.

I've been able to pull off a PBR Pipeline with effects like SSAO and the use of Ping Ponging for particle effects on both PC and Mobile, and would love to take a shot at tinkering on the switch.
